internal/install/recipe_installer_test.go

```diff
@@ -892,13 +890,85 @@ func TestInstall_GuidReport(t *testing.T) {
 	require.Equal(t, 1, statusReporters[0].(*execution.MockStatusReporter).RecipeSkippedCallCount)
 	require.Equal(t, rv.ValidateVal, statusReporters[0].(*execution.MockStatusReporter).RecipeGUID[types.InfraAgentRecipeName])
 	require.Equal(t, rv.ValidateVal, statusReporters[0].(*execution.MockStatusReporter).RecipeGUID[testRecipeName])
-	require.Equal(t, status.CLIVersion, "v0.0.1")
+	require.Equal(t, status.CLIVersion, cli.Version())
 	require.Equal(t, 3, len(statusReporters[0].(*execution.MockStatusReporter).Durations))
 	for _, duration := range statusReporters[0].(*execution.MockStatusReporter).Durations {
 		require.Less(t, int64(0), duration)
 	}
 }
 
+func TestInstall_DetectedRecipe(t *testing.T) {
+	os.Setenv("NEW_RELIC_ACCOUNT_ID", "12345")
+	ic := types.InstallerContext{}
+	statusReporters = []execution.StatusSubscriber{execution.NewMockStatusReporter()}
+	status = execution.NewInstallStatus(statusReporters, execution.NewPlatformLinkGenerator())
+	rf := recipes.NewRecipeFilterRunner(ic, status)
+	f = recipes.NewMockRecipeFetcher()
+
+	recipe := types.OpenInstallationRecipe{
+		Name:           testRecipeName,
+		ValidationNRQL: "testNrql",
+	}
+
+	f.FetchRecipesVal = []types.OpenInstallationRecipe{recipe}
+
+	v = validation.NewMockRecipeValidator()
+
+	i := RecipeInstaller{ic, d, l, mv, f, e, v, ff, status, p, pi, sp, lkf, cv, rvp, rf, av}
+
+	err := i.Install()
+
+	require.NoError(t, err)
+	require.Equal(t, 1, statusReporters[0].(*execution.MockStatusReporter).RecipeDetectedCallCount)
+}
+
+func TestInstall_DetectedMultipleRecipes(t *testing.T) {
+	os.Setenv("NEW_RELIC_ACCOUNT_ID", "12345")
+	ic := types.InstallerContext{}
+	statusReporters = []execution.StatusSubscriber{execution.NewMockStatusReporter()}
+	status = execution.NewInstallStatus(statusReporters, execution.NewPlatformLinkGenerator())
+	rf := recipes.NewRecipeFilterRunner(ic, status)
+	f = recipes.NewMockRecipeFetcher()
+
+	f.FetchRecipesVal = []types.OpenInstallationRecipe{
+		// Should NOT be detected
+		{
+			Name:           "incompatible-recipe-1",
+			ValidationNRQL: "testNrql",
+			PreInstall: types.OpenInstallationPreInstallConfiguration{
+				RequireAtDiscovery: `exit 132`,
+			},
+		},
+		// Should NOT be detected
+		{
+			Name:           "incompatible-recipe-2",
+			ValidationNRQL: "testNrql",
+			PreInstall: types.OpenInstallationPreInstallConfiguration{
+				RequireAtDiscovery: `exit 132`,
+			},
+		},
+		// Should be detected
+		{
+			Name:           "compatible-recipe-1",
+			ValidationNRQL: "testNrql",
+		},
+		// Should be detected
+		{
+			Name:           "compatible-recipe-2",
+			ValidationNRQL: "testNrql",
+		},
+	}
+
+	v = validation.NewMockRecipeValidator()
+
+	i := RecipeInstaller{ic, d, l, mv, f, e, v, ff, status, p, pi, sp, lkf, cv, rvp, rf, av}
+
+	err := i.Install()
+
+	require.NoError(t, err)
+	require.Equal(t, 2, statusReporters[0].(*execution.MockStatusReporter).RecipeDetectedCallCount)
+}
+
```

Comment:
Could we add test with process detection? A recipe with a process detected should raise the Detected status.
A recipe with process detected, and an exit code not 132 should NOT be detected.
A recipe with process detected, and no script should be detected.

Reply:
Sure thing

Reply:
Added